extension/js/common/platform/store/contact-store.ts

```diff
@@ -61,27 +70,23 @@ export class ContactStore extends AbstractStore {
   public static dbOpen = async (): Promise<IDBDatabase> => {
     return await new Promise((resolve, reject) => {
       let openDbReq: IDBOpenDBRequest;
-      openDbReq = indexedDB.open('cryptup', 3);
+      openDbReq = indexedDB.open('cryptup', 4);
       openDbReq.onupgradeneeded = (event) => {
-        let contacts: IDBObjectStore;
-        if (event.oldVersion < 1) {
-          contacts = openDbReq.result.createObjectStore('contacts', { keyPath: 'email' }); // tslint:disable-line:no-unsafe-any
-          contacts.createIndex('search', 'searchable', { multiEntry: true });
-          contacts.createIndex('index_has_pgp', 'has_pgp');
-          contacts.createIndex('index_pending_lookup', 'pending_lookup');
-        }
-        if (event.oldVersion < 2) {
-          contacts = openDbReq.transaction!.objectStore('contacts');
-          contacts.createIndex('index_longid', 'longid'); // longid of the first public key packet, no subkeys
+        if (event.oldVersion < 4) {
+          const emails = openDbReq.result.createObjectStore('emails', { keyPath: 'email' });
+          const pubkeys = openDbReq.result.createObjectStore('pubkeys', { keyPath: 'fingerprint' });
+          emails.createIndex('search', 'searchable', { multiEntry: true });
+          emails.createIndex('index_pending_lookup', 'pendingLookup');
+          emails.createIndex('index_fingerprints', 'fingerprints', { multiEntry: true }); // fingerprints of all connected pubkeys
+          pubkeys.createIndex('index_longids', 'longids', { multiEntry: true }); // longids of all public key packets in armored pubkey
         }
-        if (event.oldVersion < 3) {
-          contacts = openDbReq.transaction!.objectStore('contacts');
-          contacts.createIndex('index_longids', 'longids', { multiEntry: true }); // longids of all public key packets in armored pubkey
+        if (event.oldVersion < 5) { // delete when migrating from 4 to 5?
```

Comment:
Yep, sounds good. Leave in for now, for safety.

Reply:
In fact, I figured that we've never done db-based migration, so it would be more convenient to call some migration procedure right here. Do you have any objections against it? We can think of a nice way to inject a function here somehow.

Reply:
Sure, that's fine with me - the other migrations were for local storage which works differently. Fine to put it here.